File: python/BSTTraversal.py
class BST:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None
        
    
    def InOrder(self, array):
        if self.left != None:
            self.left.InOrder(array)
        array.append(self.value)
        if self.right != None:
            self.right.InOrder(array)
            
        return array
    
    
    def PreOrder(self, array):
        array.append(self.value)
        if self.left != None:
            self.left.PreOrder(array)
        if self.right != None:
            self.right.PreOrder(array)
            
        return array
    
    
    def PostOrder(self, array):
        if self.left != None:
            self.left.PostOrder(array)
        if self.right != None:
            self.right.PostOrder(array)
        array.append(self.value)
        
        return array
        

File: go/BSTTraversal/BSTTraversal.go
package main


type BST struct{
	Value int

	Left *BST
	Right *BST
}


func (tree *BST) InOrder(array []int) []int {
	if tree == nil {return nil}

	array = tree.Left.InOrder(array)
	array = append(array, tree.Value)
	array = tree.Right.InOrder(array)

	return array
}


func (tree *BST) PreOrder(array []int) []int {
	if tree == nil {return nil}

	array = append(array, tree.Value)
	array = tree.Left.PreOrder(array)
	array = tree.Right.PreOrder(array)

	return array
}


func (tree *BST) PostOrder(array []int) []int {
	if tree == nil {return nil}

	array = tree.Left.PostOrder(array)
	array = tree.Right.PostOrder(array)
	array = append(array, tree.Value)

	return array
}

